src/protocol/thread/thread.ts

```diff
@@ -1302,6 +1309,30 @@ class _ThreadFront {
     }
   }
 
+  private groupByContentHash(sourceIds: SourceId[]): Map<string, SourceId[]> {
+    const sourceIdsByHash = new ArrayMap<string, SourceId>();
+    for (const sourceId of sourceIds) {
+      const source = this.sources.get(sourceId);
+      assert(source, "no source found for sourceId");
+      let hash = source.contentHash;
+      // source.contentHash is not set for pretty-printed sources, we use
+      // the contentHash of the minified version instead
+      if (source.kind === "prettyPrinted") {
+        assert(
+          source.generatedSourceIds?.length === 1,
+          "a pretty-printed source should have exactly one generated source"
+        );
+        const minifiedSource = this.sources.get(source.generatedSourceIds[0]);
+        assert(minifiedSource?.contentHash, "no contentHash found for minified source");
+        hash = "minified:" + minifiedSource?.contentHash;
+      } else {
+        assert(hash, "no contentHash found for source that is not pretty-printed");
+      }
```

Comment:
These asserts feel dangerous.  If they fail, will we get problems like https://github.com/RecordReplay/devtools/issues/6547 ?  I think it would be better if we warned on getting unexpected source info from the backend, and still made a best-effort attempt to do the right thing.